packages/mobile/src/store/nativeServices/events/nativeServicesCallbacks.ts

```diff
@@ -9,10 +9,18 @@ import { navigationActions } from '../../navigation/navigation.slice'
 import { nativeServicesActions } from '../nativeServices.slice'
 
 export function* nativeServicesCallbacksSaga(): Generator {
-  const channel = yield* call(deviceEvents)
-  while (true) {
-    const action = yield* take(channel)
-    yield put(action)
+  console.log('nativeServicesCallbacksSaga starting')
+  try {
+    const channel = yield* call(deviceEvents)
+    while (true) {
+      const action = yield* take(channel)
+      yield put(action)
+    }
+  } finally {
+    console.log('nativeServicesCallbacksSaga stopping')
+    if (yield cancelled()) {
+      console.log('nativeServicesCallbacksSaga cancelled')
+    }
```

Comment:
Just adding additional logging around all of these long-running sagas (anything that runs until cancelled)